File: IT/Containers/C++/testVector1.cpp
/* 
file testVector1.cpp 
compilare con: g++ -std=c++11 testVector1.cpp
*/

#include <iostream>
#include <vector>
#include <algorithm>

using std::string;
using std::cout;
using std::endl;
using std::vector;
// using std::find;
// using std::distance;
// using std::advance;

int main()
{
	/* dichiara un vettore di stringhe */

	vector<string> vect;

	vect.push_back("tigre");
	vect.push_back("giraffa");
	vect.push_back("elefante");

	/* crea un iteratore per muoversi nel vettore */

	vector<string>::iterator it;

	/* utilizza l'iteratore per stampare gli elementi del vettore */

	for (it = vect.begin(); it != vect.end(); ++it) {
		cout << (*it) << " ";
	}

	cout << endl;

	/* verifica la presenza dell'"elefante" nella lista, se presente stampa la sua posizione */

	it = find(vect.begin(), vect.end(), "elefante");

	if (it != vect.end()) {
		int index = distance(vect.begin(), it);
		cout << "Nella lista c'è un elefante nella posizione " << index << endl;
	}

	/* modifica il secondo elemento in lista da "giraffa" a "cane"" */

	cout << "Sostituisco la giraffa con un cane." << endl;

	vect[1] = "cane";

	for (string& animal : vect) {
		cout << animal << " ";
	}

	cout << endl;

	/* possiamo aggiungere e togliere elementi da un vettore in qualsiasi momento */

	cout << "Aggiungo una balenottera azzurra." << endl;
	it = vect.begin();
	advance(it, 2);
	vect.insert(it, "balenottera");

	for (string& animal : vect) {
		cout << animal << " ";
	}

	cout << endl << "Il vettore contiene " << vect.size() << " elementi." << endl;

	/* rimuove un elemento dal vettore */

	cout << "Rimuovo il cane." << endl;

	it = find(vect.begin(), vect.end(), "cane");
	vect.erase(it);

	for (string& animal : vect) {
		cout << animal << " ";
	}

	cout << endl << "Il vettore contiene " << vect.size() << " elementi." << endl;
}

File: IT/1/Containers/Java/TestVector1.java
/* file TestVector1.java */

import java.util.Vector;

public class TestVector1 {
	public static void main(String [] args) {
		/* dichiara un vettore di stringhe */

		Vector<String> vect = new Vector<String>();

		vect.add("tigre");
		vect.add("giraffa");
		vect.add("elefante");

		/* stampa il contenuto del vettore */

		System.out.println(vect);

		/* verifica la presenza dell'"elefante" nel vettore, se presente stampa la sua posizione */

		if (vect.contains("elefante")) {
			int index = vect.indexOf("elefante");
			System.out.println("Nel vettore c'è un elefante alla posizione " + index);
		}

		/* modifica il secondo elemento del vettore: da "giraffa" a "cane" */

		System.out.println("Sostituisco la giraffa con un cane.");
		vect.set(1, "cane");
		System.out.println(vect);

		/* aggiunge una balenottera nella seconda posizione del vettore */

		System.out.println("Aggiungo una balenottera azzurra.");
		vect.add(3, "balenottera");
		System.out.println(vect);
		System.out.println("Il vettore degli animali contiene " + vect.size() + " elementi.");

		/* rimuove un elemento dal vettore */

		System.out.println("Rimuovo il cane.");
		vect.remove("cane");
		System.out.println(vect);
		System.out.println("Il vettore degli animali contiene " + vect.size() + " elementi.");
	}
}

